### PR TITLE
WAF: Move module initialization to the WAF package

### PR DESCRIPTION
**Pull request by nateweller**
#### Changes proposed in this Pull Request:

* Moves the waf module initialization into the WAF package, so that it is run by any plugin that uses `$config->ensure( 'waf' )`.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

1201069996155224-as-1203517876254355

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* TBD

### CODE REVIEW

File: projects/packages/waf/src/class-waf-initializer.php [hunk 1]
```diff
@@ -17,11 +17,25 @@ class Waf_Initializer {
...
+			return;
+		}
+
+		// Check if killswitch is defined as true
+		if ( defined( 'DISABLE_JETPACK_WAF' ) && DISABLE_JETPACK_WAF ) {
+			return;
+		}
+
+		// Triggers when the Jetpack plugin is updated
+		add_action( 'upgrader_process_complete', array( Waf_Runner::class, 'update_waf' ) );
```
Comment (dkmyta) [inline projects/packages/waf/src/class-waf-initializer.php]: This appears specific to Jetpack, do we know if this will work here as intended, or if maybe we'll need to add this line to each plugin that actually installs the WAF so the trigger actually takes effect on plugin update?

**Review by dkmyta — COMMENTED** [review]
This appears to work well for Protect and things continue to work well on the Jetpack side.

My testing process included the following:
- Activated Protect (no plan) enabled the Firewall
- Verified the request to `?test=<script>` is not blocked
- Upgraded without issue, toggled Firewall off/on, and verified that the request is now blocked
- Deactivated Protect, activated Jetpack and performed the same steps with success
- Enabled Protect in addition to Jetpack and ensured that there are no conflicts and that updates and functionality remained in parity
- Throughout each step doubled checked that all actions remained functional
- Tested using Jurassic Tube to cover all changes
- Tested using Jurassic Ninja and Pressable w/ the Jetpack Beta tester and this branch to cover all things Protect related

Is there anything else critical to consider testing that wouldn't have been covered in the above steps with regard to plugin (Jetpack or Protect) or package functionality or the various processes involved?

One other thought came to mind, will this all be backward compatible? What if, for example, someone has the latest version of Protect installed and active, but a version of Jetpack active from prior to the change - could this potentially cause any issues with WAF functionality, or in-plugin settings changes? Testing didn't seem to indicate any issues, but figured it's worth ensuring we have it covered.

File: projects/packages/waf/src/class-waf-initializer.php (hunk 1 above)
Comment (jwidavid) [reply projects/packages/waf/src/class-waf-initializer.php]: That hook should work regardless of the plugin, although the comment could probably be updated.

File: projects/plugins/protect/.gitattributes [hunk 2]
```diff
@@ -13,6 +13,7 @@ vendor/autoload_packages.php                    production-include
 vendor/automattic/**                            production-include
 vendor/composer/**                              production-include
 vendor/jetpack-autoloader/**                    production-include
+vendor/wikimedia/aho-corasick/**                production-include
```
Comment (jwidavid) [inline projects/plugins/protect/.gitattributes:16]: Why is this being added?

Comment (dkmyta) [reply projects/plugins/protect/.gitattributes:16]: The discussion can be found here: p1670548960264089-slack-CBG1CP4EN

TL;DR as I understood it, that package dependency wasn't being added to Protect's list of dependencies and it was causing a fatal somewhere down the line.

File: projects/packages/waf/src/class-waf-initializer.php (hunk 1 above)
Comment (nateweller) [reply projects/packages/waf/src/class-waf-initializer.php]: Looking into the action details, this runs any time anything in WordPress updates. If we only want to run this when one of our plugins are updated, I can add a wrapper function that only runs the `update_waf` method if we just updated the plugin that is currently using the package.

Comment (nateweller) [reply projects/packages/waf/src/class-waf-initializer.php]: Like so: 04f6fac46c8e

**Comment by nateweller** [comment]
@dkmyta @jwidavid I've addressed all comments, if you could peruse the one additional change to the `upgrader_process_complete` hook :+1:

Honestly best way I've found to test it is `error_log`s to ensure it is executing at the right time 😅

**Review by dkmyta — APPROVED** [review]
Works like a charm!

**Review by sdixon194 — APPROVED** [review]
Looks like the module still loads from Jetpack's end 👍
